scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -200,11 +200,15 @@ trait Config extends Serializable {
 
   def getArgs: Args = get(Config.ScaldingJobArgs) match {
     case None => new Args(Map.empty)
-    case Some(str) => Args(str)
+    case Some(str) => argsSerializer
+      .invert(str)
+      .map(new Args(_))
+      .getOrElse(throw new RuntimeException(
+        s"""Could not deserialize Args from Config. Maybe "$ScaldingJobArgs" was modified without using Config.setArgs?"""))
   }
 
   def setArgs(args: Args): Config =
-    this + (Config.ScaldingJobArgs -> args.toString)
+    this + (Config.ScaldingJobArgs -> argsSerializer(args.m))
```

Comment:
can we put this in a new key and keep the old one for people that might what to read those keys?

It it nice for human debugging and cheap to add a new key.